test/extended/apiserver/graceful_termination.go

```diff
@@ -143,84 +145,109 @@ var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Late]", func() {
 	})
 
 	g.It("API LBs follow /readyz of kube-apiserver and stop sending requests before server shutdowns for external clients", func() {
-		// set up
-		results := map[string]int{}
-		tempDir, err := ioutil.TempDir("", "test.oc-adm-must-gather.")
+		controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
 		o.Expect(err).NotTo(o.HaveOccurred())
-		defer os.RemoveAll(tempDir)
-		args := []string{"--dest-dir", tempDir, "--", "/usr/bin/gather_audit_logs"}
-
-		// download the audit logs from the cluster
-		o.Expect(oc.Run("adm", "must-gather").Args(args...).Execute()).To(o.Succeed())
-
-		// act
-		expectedDirectoriesToExpectedCount := []string{path.Join(clihelpers.GetPluginOutputDir(tempDir), "audit_logs", "kube-apiserver")}
-		for _, auditDirectory := range expectedDirectoriesToExpectedCount {
-			err := filepath.Walk(auditDirectory, func(path string, info os.FileInfo, err error) error {
-				g.By(path)
-				o.Expect(err).NotTo(o.HaveOccurred())
 
-				if info.IsDir() {
-					return nil
-				}
-				fileName := filepath.Base(path)
-				if !clihelpers.IsAuditFile(fileName) {
-					return nil
-				}
+		// set up
+		// apiserverName -> reader
+		logReaders := map[string]io.Reader{}
+		results := map[string]int{}
 
-				file, err := os.Open(path)
+		if *controlPlaneTopology == configv1.ExternalTopologyMode {
+			mgmtClusterOC := exutil.NewHypershiftManagementCLI("default").AsAdmin().WithoutNamespace()
+			pods, err := mgmtClusterOC.KubeClient().CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: "hypershift.openshift.io/control-plane-component=kube-apiserver"})
+			o.Expect(err).To(o.BeNil())
+			for _, pod := range pods.Items {
+				fileName, err := mgmtClusterOC.Run("logs", "-n", pod.Namespace, pod.Name, "-c", "audit-logs").OutputToFile(pod.Name + "-audit.log")
```

Comment:
I think it only gets the latest logs, during e2e tests APIs might be restarted multiple times. The intention of this test was to validate that during a termination we didn't receive late requests. Checking all available logs increases our chances to detect incorrect behavior.

I think we would have to list all available logs and retrieve them individually. 

I am okay with what we have now we could just improve it in the future.

Reply:
>  APIs might be restarted multiple times. 

Why would the api get restarted during e2e tests?

Reply:
some tests do that, i.e. `etcd is able to vertically scale up and down with a single node`